lib/tls.js

```diff
@@ -75,7 +75,7 @@ exports.checkServerIdentity = function checkServerIdentity(host, cert) {
   // Create regexp to much hostnames
   function regexpify(host, wildcards) {
     // Add trailing dot (make hostnames uniform)
-    if (!/\.$/.test(host)) host += '.';
+    if (!host || !host.endsWith('.')) host += '.';
```

Comment:
Well, it wouldn't have blown up, but would've turned `undefined` into `"undefined"`, before appending a period to it. My point is, this was never a bug, so do we need that `!host` check?